File: floating-clock-widget/native/FloatingClockWidget.Native/Services/RuntimePaths.cs
using System;
using System.IO;

namespace FloatingClockWidget.Native.Services;

public sealed record RuntimePaths(
    string RootDir,
    string ThemesDir,
    string ImportsTmpDir,
    string ImportsFailedDir,
    string LogsDir)
{
    public static RuntimePaths Resolve()
    {
        var localAppData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
        var root = Path.Combine(localAppData, "FloatingClockWidget");

        return new RuntimePaths(
            RootDir: root,
            ThemesDir: Path.Combine(root, "themes"),
            ImportsTmpDir: Path.Combine(root, "imports", "tmp"),
            ImportsFailedDir: Path.Combine(root, "imports", "failed"),
            LogsDir: Path.Combine(root, "logs"));
    }
}

File: floating-clock-widget/src/theme/runtimePaths.js
const path = require("node:path");

const APP_RUNTIME_FOLDER = "FloatingClockWidget";

function resolveRuntimePaths(localAppDataPath) {
  const rootDir = path.join(localAppDataPath, APP_RUNTIME_FOLDER);

  return {
    rootDir,
    themesDir: path.join(rootDir, "themes"),
    importsTmpDir: path.join(rootDir, "imports", "tmp"),
    importsFailedDir: path.join(rootDir, "imports", "failed"),
    logsDir: path.join(rootDir, "logs"),
  };
}

module.exports = {
  APP_RUNTIME_FOLDER,
  resolveRuntimePaths,
};
